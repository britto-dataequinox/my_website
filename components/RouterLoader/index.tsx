import React from 'react';
import Head from 'next/head';
import CircularProgress from '@mui/material/CircularProgress';
import { Box, Typography } from '@mui/material';

const RouterLoader = () => {
  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', height: '100vh' }}>
      <Head>
        <title>Loading | BTM INDUSTRIES</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Box sx={{ position: 'relative', display: 'inline-flex', alignItems: 'center' }}>
        <CircularProgress data-testid='router-loader' size={150} />
        <Typography
          variant="body1"
          component="div"
          sx={{
            position: 'absolute',
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)',
            color: 'rgba(0, 0, 0, 0.5)',
          }}
        >
          BTM INDUSTRIES
        </Typography>
      </Box>
    </Box>
  );
};

export default RouterLoader;
