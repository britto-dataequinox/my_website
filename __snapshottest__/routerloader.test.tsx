import React from 'react';
import renderer from 'react-test-renderer';
import RouterLoader from '../components/RouterLoader';

test('router loader snap', () => {
  const tree = renderer.create(<RouterLoader />).toJSON();
  expect(tree).toMatchSnapshot();
});
