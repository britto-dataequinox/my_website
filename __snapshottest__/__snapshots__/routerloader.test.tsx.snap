// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router loader snap 1`] = `
<div
  className="MuiBox-root css-65avyl"
>
  <div
    className="MuiBox-root css-datla1"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      data-testid="router-loader"
      role="progressbar"
      style={
        {
          "height": 150,
          "width": 150,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={{}}
        />
      </svg>
    </span>
    <div
      className="MuiTypography-root MuiTypography-body1 css-1e0k6sq-MuiTypography-root"
    >
      BTM INDUSTRIES
    </div>
  </div>
</div>
`;
